### PR TITLE
Feature/news scrapper

### PR DESCRIPTION
**Pull request by Hediby**
Create a basic scraper for Google News. This scraper uses the RSS feed for Google News and calls basic urls:
- Top news
- News per topic, on the few existing topics in Google News.

The parser also implements a method to retrieve article urls that satisfy a certain query, but this method is not used yet. 

- [x] Create the GoogleNewsScraper class
- [x] Write a script that uses a GoogleNewsScraper object
- [ ] Write tests

### CODE REVIEW

File: src/scraper.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import feedparser
+
+AUTHORIZED_TOPICS = [
```
Comment (nicolaslecrique) [inline src/scraper.py:4]: it's more of a set of "selected" topics than authorized (we load all top news on top of it that may not be in authorized

File: src/scraper.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "WORLD",
+    "NATION",
+    "BUSINESS",
+    "TECHNOLOGY",
+    "ENTERTAINMENT",
+    "SPORTS",
+    "SCIENCE",
+]
```
Comment (nicolaslecrique) [inline src/scraper.py:13]: is there a link to a reference to all valid topics from google news ?

File: src/scraper.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        query_suffix = self.query_suffix_template.format(query=query)
+        url = self.base_url + query_suffix + "&" + self.lang_suffix
+        retrieved_urls = self.retrieve_urls(url)
+        return retrieved_urls
+
+
+    def top_news_from_topic(self, topic: str) -> list[str]:
+        # topic in WORLD NATION BUSINESS TECHNOLOGY ENTERTAINMENT SPORTS SCIENCE 
+        assert topic in AUTHORIZED_TOPICS
+        url = self.topic_url_template.format(topic = topic) + self.lang_suffix
```
Comment (nicolaslecrique) [inline src/scraper.py]: i would rename to topic_url to not mix it up with article urls

File: src/scraper.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        retrieved_urls = self.retrieve_urls(url)
+        return retrieved_urls
+    
+    def retrieve_urls(self, url: str) -> list[str]:
+        response = feedparser.parse(url)
+        output_urls = []
+        for entry in response['entries']:
+            output_urls.append(entry['link'])
+        return output_urls
```
Comment (nicolaslecrique) [inline src/scraper.py]: I think you can use a list comprehension like

```suggestion
        return [entry['link'] for entry in response['entries']]
```

File: src/scraper.py (hunk 2 above)
Comment (Hediby) [reply src/scraper.py:13]: Here is the only info I found: https://stackoverflow.com/questions/51537063/url-format-for-google-news-rss-feed

File: src/scraper.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        self.topic_url_template = self.base_url + "/" + "headlines/section/topic/{topic}"
+
+
+    def top_news(self) -> list[str]:
+        url = self.base_url + self.lang_suffix
+        retrieved_urls = self.retrieve_urls(url)
+        return retrieved_urls
+        
+
+    def query(self, query: str) -> list[str]:
```
Comment (nicolaslecrique) [inline src/scraper.py]: dead code ? i don't see the usage

File: src/scraper.py (hunk 1 above)
Comment (Hediby) [reply src/scraper.py:4]: The topics are quite generic and fixed, not every topic is available. This list corresponds to the banner at the top of Google News
image

File: src/scraper.py (hunk 3 above)
Comment (Hediby) [reply src/scraper.py]: Done

File: src/scraper.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+
+    def top_news_from_topic(self, topic: str) -> list[str]:
+        # topic in WORLD NATION BUSINESS TECHNOLOGY ENTERTAINMENT SPORTS SCIENCE 
+        assert topic in AUTHORIZED_TOPICS
+        url = self.topic_url_template.format(topic = topic) + self.lang_suffix
+        retrieved_urls = self.retrieve_urls(url)
+        return retrieved_urls
+    
+    def retrieve_urls(self, url: str) -> list[str]:
```
Comment (nicolaslecrique) [inline src/scraper.py]: you could remove self as it's not used (or even move it like an "utils" function outside of the class) as it's basically a wrapper around feedparser and note related to google news

File: src/scraper.py (hunk 2 above)
Comment (Hediby) [reply src/scraper.py:13]: I added a comment with the link.

File: src/scraper.py (hunk 4 above)
Comment (Hediby) [reply src/scraper.py]: Done

File: src/scraper.py (hunk 6 above)
Comment (Hediby) [reply src/scraper.py]: Done

File: src/scraper.py (hunk 5 above)
Comment (Hediby) [reply src/scraper.py]: Not used yet, but we could design a more elaborated `scripts/scraper.py` that uses this method. Should I remove it ?

Comment (nicolaslecrique) [reply src/scraper.py]: following YAGNI principle, i woule say that we can add it back if and when we need it
https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

**Review by nicolaslecrique — APPROVED** [review]

**Pull request merged**